### PR TITLE
Story 23

### PR DESCRIPTION
**Pull request by agallant121**
Complete user story 23

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -22,12 +22,30 @@ def remove_item
     redirect_to '/cart'
   end
 
+  def edit_quantity
+    adjust_quantity
```
Comment (agallant121) [inline app/controllers/cart_controller.rb:26]: Put in helper method for the functionality which is now down in the private section under adjust_quantity

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -22,12 +22,30 @@ def remove_item
...
   def require_not_admin
       render file: "/public/404" if current_admin
   end
 
+  def item
+    @item = Item.find(params[:item_id])
+  end
+
+  def adjust_quantity
+    if params[:quantity] == "add" && item.inventory > session[:cart][params[:item_id]]
```
Comment (agallant121) [inline app/controllers/cart_controller.rb:40]: This takes the quantity param of an item and clicks add to increment quantity by one for each specific item but if the number of 'adds' goes beyond what the set quantity is available you get a flash message stating the item is out of stock

File: app/controllers/cart_controller.rb [hunk 3]
```diff
@@ -22,12 +22,30 @@ def remove_item
...
+      session[:cart][params[:item_id]] += 1
+      redirect_to '/cart'
```
Comment (agallant121) [inline app/controllers/cart_controller.rb:42]: This was all based off the code left behind at the bottom that was commented out

File: app/views/cart/show.html.erb [hunk 4]
```diff
@@ -30,7 +30,8 @@
       </td>
       <td>
         <center>
-          <p><%=quantity%></p>
+          <p id = "item-quantity-<%=item.id%>"><%=quantity%></p>
```
Comment (agallant121) [inline app/views/cart/show.html.erb:33]: Added an id to be able to test within the item's quantity section to make sure the numbers we test for are here and not elsewhere on the page

File: app/views/cart/show.html.erb [hunk 5]
```diff
@@ -30,7 +30,8 @@
...
+          <p><%= link_to "+", "/cart/#{item.id}/add",method: :patch %>
```
Comment (agallant121) [inline app/views/cart/show.html.erb:34]: Using the + sign to increment quantity of specific items

File: config/routes.rb [hunk 6]
```diff
@@ -30,6 +30,8 @@
   get "/cart", to: "cart#show"
   delete "/cart", to: "cart#empty"
   delete "/cart/:item_id", to: "cart#remove_item"
+  patch "/cart/:item_id/:quantity", to: "cart#edit_quantity"
```
Comment (agallant121) [inline config/routes.rb:33]: new route to the cart controller to edit the quantity of a specific item

File: spec/features/cart/increment_cart_spec.rb [hunk 7]
```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+RSpec.describe 'Cart inrementation' do
+    before(:each) do
+      @mike = Merchant.create(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
```
Comment (agallant121) [inline spec/features/cart/increment_cart_spec.rb:5]: Story 23 Spec; This spec goes through and adds items and makes sure we are on the cart view before testing. We add the paper item until it is beyond what is in stock and make sure that it sends a flash message if you try to order too many

**Review by meghanstovall — APPROVED** [review]

**Pull request merged**